docs/dataset/introduction.rst

```diff
@@ -0,0 +1,69 @@
+
+For an example notebook showing the useage of the database as a container for experiments, see the :doc:`Experiment Container Notebook <../_notebooks/DataSet/example-creation>`.
+
+The intended use of the ``Experiment`` is to group ``DataSets`` that belong together. For example, an experiment may contain all the measurements necessary to characterise a particular sample (Is it superconducting? What is the critical current?). Once the goodness of the sample has been established, a new experiment containing all the "real" measurements can then commence. The grouping runs this way is offered as a convenience and is as such not essential for any functionality in QCoDeS.
+
+.. _sec:intro_dataset:
+
+DataSet
+=======
+
+Each ``DataSet`` has the following attributes
+
+  * A run-ID
+  * An experiment-ID
+  * A name
+  * A table of results
+  * A number of parameters
+  * A start time
+  * (potentially) An end time
+  * A GUID
+
+The **run-ID** is an ever-increasing integer enumerating the number of runs in the current database. The runs are kept in the SQLite database in the ``runs`` table. The run-ID is the primary key of this table.
+
+The **experiment-ID** provides the link to the experiment that this dataset belongs to.
+
+The **name** must be provided by the user at creation time.
+
+The **table of results** is where the actual *data* of the ``DataSet`` resides. The table has a number of associated **parameters** as columns with data points as rows. The parameters may be QCoDeS ``Parameters``, but are not limited to that. The preferred procedure for associating parameters with a run and adding data is via the ``Measurement`` object as described in the :doc:`Measurement Object Example Notebook <../_notebooks/DataSet/Dataset Context Manager>`. The deep thinking behind how different parameters relate to and depend on each other within a ``DataSet`` is explained in :ref:`interdependentparams`.
+
+The **start time** is automatically added upon creation, and the **end time** is added when (if ever) the user decides to mark the run as *completed* after which point no more data points may be added.
+
+The **GUID** is a run identifier that aims to go `beyond` the current database. The motivation for the GUID is that collaboration and data sharing across several machines requires a run to be labelled by something more than its name and number in the local database. The GUID is a standard 36 character string GUID composed from three integer codes and a timestamp. The codes are: location code (1-256), work station code (1-16777216), sample code (1-4294967296). The idea is that the people in a certain collaboration will label their locations, machines, and samples according to a scheme agreed upon `within` the relevant collaboration. That way, each run is uniquely identifiable in a human-readable way. This extra label for runs is offered as a convenience as is as such not essential for any functionality in QCoDeS.
+
```

Comment:
do we need to document `generate_guid`, `parse_guid`, `set_guid_*`, `update_GUIDs` functions anywhere for someone?

i'm just checking. i thing that these should only be known to developers, and should not be used by users.